src/components/data-list/index.tsx

```diff
@@ -86,18 +42,24 @@ export const DataList = ({
   const [skipLimit, setSkipLimit] = useState(0);
   const [totalRowCount, setTotalRowCount] = useState(0);
   const [pageNumber, setPageNumber] = useState(0);
-  const [isImportWindowOpen, setIsImportWindowOpen] = useState(false);
-  const [openExport, setOpenExport] = useState(false);
   const [columnVisibilityModel, setColumnVisibilityModel] = useState(
     initialGridState?.columns?.columnVisibilityModel
   );
   const [gridSettingsLoaded, setGridSettingsLoaded] = useState(false);
+  const [gridState, setGridState] = useState(initialGridState);
 
   const whereFilterQuery = getWhereFilterQuery(whereField, whereFieldValue);
   const basicFilterQuery = getBasicFilterQuery(filterLimit, sortColumn, sortOrder, skipLimit);
   const basicExportFilterQuery = getBasicExportFilterQuery(sortColumn, sortOrder);
 
-  const gridSettingsStorageName = `${modelName}DataListSettings`;
+  const [gridSettings, setGridSettings] = useLocalStorage<dataListSettings | undefined>(
+    gridSettingsStorageKey,
+    undefined
+  );
+
+  useEffect(() => {
+    setSearchTerm(searchText);
+  }, [searchText]);
 
   useEffect(() => {
     if (gridSettingsLoaded) {
```

Comment:
look into gridSettingsLoaded using logic if it realy necessary

Reply:
![image](https://user-images.githubusercontent.com/3434811/233555462-cf425626-e1b6-46dd-95e4-a756aa1d2961.png)